amethyst_utils/src/auto_fov.rs

```diff
@@ -19,35 +19,38 @@ use serde::{Deserialize, Serialize};
 use thread_profiler::profile_scope;
 
 /// A component describing the behavior of the camera in accordance with the screen dimensions
-#[derive(Clone, Debug, Deserialize, PrefabData, Serialize)]
+#[derive(Clone, Debug, Deserialize, PrefabData, Serialize, new, Setters)]
 #[prefab(Component)]
 #[serde(default)]
 pub struct AutoFov {
     /// The horizontal FOV value at the aspect ratio in the field `base_aspect_ratio`
+    #[new(default)]
     base_fovx: f32,
 
+    /// Sets `fovx_growth_rate` to the given value
+    #[set = "pub"]
     /// The factor determining how sensitive the FOV change should be
+    #[new(default)]
     fovx_growth_rate: f32,
 
     /// If the FOV grow rate specified in the field `fovx_growth_rate` should be applied as-is
+    #[new(default)]
     fixed_growth_rate: bool,
 
     /// The aspect ratio when the camera's horizontal FOV is identical to `base_fovx`
+    #[new(default)]
     base_aspect_ratio: (usize, usize),
 
     /// The minimum value the horizontal FOV can have
+    #[new(default)]
     min_fovx: f32,
```

Comment:
If you look at the Default implementation, you'll see that there's default non-zero values for those. :)

Reply:
Yeah, my bad